src/Components/Endpoints/src/Rendering/SSRRenderModeBoundary.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.Web;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Components.Endpoints;
+
+/// <summary>
+/// A component that describes a location in prerendered output where client-side code
+/// should insert an interactive component.
+/// </summary>
+internal class SSRRenderModeBoundary : IComponent
```

Comment:
Having the boundary be an actual component in the hierarchy means:

 * It automatically captures the parameters that we need to pass to the interactive child, without any extra logic in the prerendering or rendering systems
 * We can encapsulate things like "doing prerendering or not" with a trivial "if" condition here
 * In the future, if we want to extend to something like blocking nested interactive components, we can do so trivially using cascading parameters without complexifying anything in the renderer